src/fu-progressbar.c

```diff
@@ -135,6 +138,16 @@ fu_progressbar_refresh (FuProgressbar *self, FwupdStatus status, guint percentag
 	}
 	g_string_append_c (str, ']');
 
+	/* once we have some data (10%) show an estimate of time remaining */
+	if (percentage < 100 && percentage > 10) {
+		gdouble rate = percentage / g_timer_elapsed (self->time_elapsed, NULL);
+		time_remaining = g_strdup_printf ("%.2f", (100-percentage) * rate);
+		g_string_append_c (str, ' ');
+		g_string_append (str, _("Estimated time remaining:"));
```

Comment:
Needs a translator comment explaining the context

Reply:
added